### PR TITLE
feat: configure linkify in markdownit with tlds

### LINKED ISSUE #2304: .rocks domains do not get auto-linkified
**Which part of the project should be enhanced?**
The...renderer? auto-linkifier?

**Is your enhancement request related to a problem? Please describe.**
If I add plain links to my Markdown document using TLDs like .de or .com, they automatically are converted into links. The .rocks TLD is not auto-converted:

image

**Describe the solution you'd like**
All links with domains from the [public suffix list](https://publicsuffix.org/list/public_suffix_list.dat) should be auto-linkified.

**Describe alternatives you've considered**
Keep the current system that provides a confusing user-experience.

**Additional context**

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
renderer

### Description
This PR fixes the autodetection of linkify.

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### Related Issue(s)
closes #2304

### CODE REVIEW

**Comment by DerMolly** [comment]
grafik

File: package.json [hunk 1]
```diff
@@ -120,6 +120,7 @@
     "sanitize-filename": "1.6.3",
     "sass": "1.54.5",
     "sharp": "0.30.7",
+    "tlds": "^1.231.0",
```
Comment (mrdrogdrog) [inline package.json:123]: Pin dependency

**Review by mrdrogdrog — CHANGES_REQUESTED** [review]

**Review by mrdrogdrog — CHANGES_REQUESTED** [review]
Something here isn't right.

You added the fix to the rendering hook but are testing it in the test file of an extension.

I suggest that you move the fix into the linkify extension. That makes more sense than putting more stuff into the rendering test.

**Comment by DerMolly** [comment]
Good point. I think this is better?

**Review by mrdrogdrog — APPROVED** [review]

**Pull request merged**